src/collectives/device/primitives.h

```diff
@@ -44,11 +44,13 @@ class PostFlag {
   const int shift;
   volatile int * const fifo;
   const int fifo_size;
+  uint32_t * hdp_reg;
  public:
   __device__
-  PostFlag(volatile uint64_t* const flag, const int shift, volatile int* const fifo, const int fifo_size) : flag(flag), shift(shift), fifo(fifo), fifo_size(fifo_size) { }
+  PostFlag(volatile uint64_t* const flag, const int shift, volatile int* const fifo, const int fifo_size, uint32_t* hdp_reg = NULL)
+    : flag(flag), shift(shift), fifo(fifo), fifo_size(fifo_size), hdp_reg(hdp_reg) { }
   __device__
-  void post(uint64_t val) { STORE(flag, (val - shift)); }
+  void post(uint64_t val) { if (hdp_reg != NULL) STORE(hdp_reg, 0x1); STORE(flag, (val - shift)); }
```

Comment:
This is rather mysterious.  Can you provide a comment which explains what this ends up doing?

Reply:
Done. The descriptions were given by Sean from his earlier email.